libraries/AP_HAL_Linux/RPIOUARTDriver.cpp

```diff
@@ -162,26 +163,17 @@ void RPIOUARTDriver::_timer_tick(void)
     struct IOPacket _dma_packet_tx = {0}, _dma_packet_rx = {0};
 
     /* get write_buf bytes */
-    uint32_t n = _writebuf.available();
-
-    if (n > PKT_MAX_REGS * 2) {
-        n = PKT_MAX_REGS * 2;
-    }
-
-    uint16_t _max_size = _baudrate / 10 / (1000000 / RPIOUART_POLL_TIME_INTERVAL);
-    if (n > _max_size) {
-        n = _max_size;
-    }
+    uint32_t n = MIN(_writebuf.available(), PKT_MAX_REGS * 2);
+    n = MIN(n, _baudrate / 10U / (1000000U / RPIOUART_POLL_TIME_INTERVAL));
 
     _writebuf.read(&((uint8_t *)_dma_packet_tx.regs)[0], n);
 
     _dma_packet_tx.count_code = PKT_MAX_REGS | PKT_CODE_SPIUART;
     _dma_packet_tx.page = PX4IO_PAGE_UART_BUFFER;
     _dma_packet_tx.offset = n;
+    _dma_packet_tx.crc = crc_packet(&_dma_packet_tx);
```

Comment:
But you are right. In one of the cases we were reusing the struct and not zero'ing it, my bad.